### PR TITLE
Fix docstrings on torch/optim and torch/autograd

### LINKED ISSUE #112593: Fix docstring errors in nadam.py, radam.py, sgd.py, anomaly_mode.py, rprop.py, __init__.py, swa_utils.py, rmsprop.py, optimizer.py, lr_scheduler.py
Please fix the following issues. To test locally, run: 
 ```
pydocstyle path-to-file --count
```
Run this command before and after fixing the errors. This command prints out the number of errors at the end of the output. Paste the output to your PR description with the number before and after. 
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `_initial_step`, **Line**: 82, **Description**: First line should end with a period (not 'p')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 88, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 96, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_last_lr`, **Line**: 105, **Description**: One-line docstring should fit on one line with quotes (found 2)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_last_lr`, **Line**: 105, **Description**: No whitespaces allowed surrounding docstring text
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `print_lr`, **Line**: 114, **Description**: One-line docstring should fit on one line with quotes (found 2)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `LambdaLR`, **Line**: 187, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `LambdaLR`, **Line**: 187, **Description**: First line should end with a period (not 'r')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 224, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 224, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 244, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 244, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `MultiplicativeLR`, **Line**: 273, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `MultiplicativeLR`, **Line**: 273, **Description**: First line should end with a period (not 'n')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 308, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 325, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `StepLR`, **Line**: 354, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `StepLR`, **Line**: 354, **Description**: First line should end with a period (not 'y')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `MultiStepLR`, **Line**: 403, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `MultiStepLR`, **Line**: 403, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ConstantLR`, **Line**: 452, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ConstantLR`, **Line**: 452, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `LinearLR`, **Line**: 510, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `LinearLR`, **Line**: 510, **Description**: First line should end with a period (not 'l')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ExponentialLR`, **Line**: 578, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `SequentialLR`, **Line**: 609, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `SequentialLR`, **Line**: 609, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 686, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 701, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `PolynomialLR`, **Line**: 718, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `PolynomialLR`, **Line**: 718, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `PolynomialLR`, **Line**: 718, **Description**: First line should end with a period (not 'n')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CosineAnnealingLR`, **Line**: 768, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CosineAnnealingLR`, **Line**: 768, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ChainedScheduler`, **Line**: 841, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ChainedScheduler`, **Line**: 841, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 882, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 897, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ReduceLROnPlateau`, **Line**: 914, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `_reset`, **Line**: 1002, **Description**: First line should be in imperative mood (perhaps 'Reset', not 'Resets')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CyclicLR`, **Line**: 1088, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CyclicLR`, **Line**: 1088, **Description**: First line should end with a period (not 'o')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_lr`, **Line**: 1285, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_lr`, **Line**: 1285, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_lr`, **Line**: 1285, **Description**: First line should end with a period (not 's')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_lr`, **Line**: 1285, **Description**: First line should be in imperative mood (perhaps 'Calculate', not 'Calculates')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CosineAnnealingWarmRestarts`, **Line**: 1339, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CosineAnnealingWarmRestarts`, **Line**: 1339, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `step`, **Line**: 1390, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `step`, **Line**: 1390, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `OneCycleLR`, **Line**: 1464, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `OneCycleLR`, **Line**: 1464, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `OneCycleLR`, **Line**: 1464, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `_annealing_cos`, **Line**: 1695, **Description**: Use """triple double quotes""" (found "-quotes)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `_annealing_linear`, **Line**: 1701, **Description**: Use """triple double quotes""" (found "-quotes)
- **File**: `torch/optim/nadam.py`, **Entity**: `step`, **Line**: 70, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/nadam.py`, **Entity**: `nadam`, **Line**: 176, **Description**: No blank lines allowed after function docstring (found 2)
- **File**: `torch/optim/optimizer.py`, **Entity**: `_RequiredParameter`, **Line**: 25, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_optimizer_step_pre_hook`, **Line**: 143, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_optimizer_step_pre_hook`, **Line**: 143, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_optimizer_step_post_hook`, **Line**: 162, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_optimizer_step_post_hook`, **Line**: 162, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/optimizer.py`, **Entity**: `_group_tensors_by_device_and_dtype`, **Line**: 323, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `_group_tensors_by_device_and_dtype`, **Line**: 323, **Description**: Multi-line docstring closing quotes should be on a separate line
- **File**: `torch/optim/optimizer.py`, **Entity**: `_group_tensors_by_device_and_dtype`, **Line**: 323, **Description**: First line should be in imperative mood (perhaps 'Group', not 'Groups')
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_step_pre_hook`, **Line**: 338, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_step_pre_hook`, **Line**: 338, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_step_post_hook`, **Line**: 360, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `state_dict`, **Line**: 380, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/optimizer.py`, **Entity**: `load_state_dict`, **Line**: 432, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/optimizer.py`, **Entity**: `zero_grad`, **Line**: 488, **Description**: First line should be in imperative mood (perhaps 'Reset', not 'Resets')
- **File**: `torch/optim/optimizer.py`, **Entity**: `step`, **Line**: 529, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/optimizer.py`, **Entity**: `step`, **Line**: 529, **Description**: First line should not be the function's "signature"
- **File**: `torch/optim/radam.py`, **Entity**: `step`, **Line**: 84, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/radam.py`, **Entity**: `radam`, **Line**: 200, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/rmsprop.py`, **Entity**: `step`, **Line**: 97, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/rmsprop.py`, **Entity**: `rmsprop`, **Line**: 217, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/rmsprop.py`, **Entity**: `rmsprop`, **Line**: 217, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/rprop.py`, **Entity**: `step`, **Line**: 82, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/rprop.py`, **Entity**: `rprop`, **Line**: 188, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/sgd.py`, **Entity**: `step`, **Line**: 57, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/sgd.py`, **Entity**: `sgd`, **Line**: 200, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/swa_utils.py`, **Entity**: `AveragedModel`, **Line**: 67, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/swa_utils.py`, **Entity**: `AveragedModel`, **Line**: 67, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/swa_utils.py`, **Entity**: `AveragedModel`, **Line**: 67, **Description**: First line should end with a period (not 'd')
- **File**: `torch/optim/swa_utils.py`, **Entity**: `update_bn`, **Line**: 227, **Description**: First line should be in imperative mood (perhaps 'Update', not 'Updates')
- **File**: `torch/optim/swa_utils.py`, **Entity**: `SWALR`, **Line**: 280, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/_multi_tensor/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/autograd/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/autograd/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not 'c')
- **File**: `torch/autograd/__init__.py`, **Entity**: `backward`, **Line**: 118, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/autograd/__init__.py`, **Entity**: `backward`, **Line**: 118, **Description**: First line should end with a period (not 'h')
- **File**: `torch/autograd/__init__.py`, **Entity**: `backward`, **Line**: 118, **Description**: First line should be in imperative mood (perhaps 'Compute', not 'Computes')
- **File**: `torch/autograd/__init__.py`, **Entity**: `grad`, **Line**: 219, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/autograd/__init__.py`, **Entity**: `grad`, **Line**: 219, **Description**: First line should end with a period (not 'o')
- **File**: `torch/autograd/__init__.py`, **Entity**: `grad`, **Line**: 219, **Description**: First line should be in imperative mood (perhaps 'Compute', not 'Computes')
- **File**: `torch/autograd/anomaly_mode.py`, **Entity**: `detect_anomaly`, **Line**: 10, **Description**: No blank lines allowed between a section header and its content ('Example')

cc @brycebortree @carljparker

### PR DESCRIPTION
**Pull request by alperenunlu**
Fixes #112593

## Description
Fixes the docstrings on following files.

```bash
pydocstyle path-to-file --count
```
| File                                  |  Count  |
| ------------------------------------- | ------- |
| torch/optim/lr_scheduler.py           | 96 -> 39|
| torch/optim/nadam.py                  | 6 -> 4 |
| torch/optim/optimizer.py              | 31 -> 6|
| torch/optim/radam.py                  | 6 -> 4 |
| torch/optim/rmsprop.py                | 7 -> 4 |
| torch/optim/rprop.py                  | 6 -> 4 |
| torch/optim/sgd.py                    | 6 -> 4 |
| torch/optim/swa_utils.py              | 15 -> 10|
| torch/optim/_multi_tensor/__init__.py | 2 -> 1 |
| torch/autograd/__init__.py            | 8 -> 0 |
| torch/autograd/anomaly_mode.py        | 8 -> 7 |

### CODE REVIEW

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -444,9 +446,12 @@ def _get_closed_form_lr(self):
 
 
 class ConstantLR(LRScheduler):
-    """Decays the learning rate of each parameter group by a small constant factor until the
-    number of epoch reaches a pre-defined milestone: total_iters. Notice that such decay can
-    happen simultaneously with other changes to the learning rate from outside this scheduler.
+    """Decays the learning rate of each parameter group by a small constant factor.
+
+    Until the number of epoch reaches a pre-defined milestone: total_iters.
```
Comment (svekars) [inline torch/optim/lr_scheduler.py:451]:
```suggestion
    """Decays the learning rate of each parameter group by a small constant factor until the number of epoch reaches a pre-defined milestone: total_iters.
```

File: torch/optim/lr_scheduler.py [hunk 2]
```diff
@@ -500,8 +505,10 @@ def _get_closed_form_lr(self):
 
 
 class LinearLR(LRScheduler):
-    """Decays the learning rate of each parameter group by linearly changing small
-    multiplicative factor until the number of epoch reaches a pre-defined milestone: total_iters.
+    """Decays the learning rate of each parameter group by linearly changing small multiplicative factor.
+
+    Until the number of epoch reaches a pre-defined milestone: total_iters.
```
Comment (svekars) [inline torch/optim/lr_scheduler.py:510]:
```suggestion
    """Decays the learning rate of each parameter group by linearly changing small multiplicative factor until the number of epoch reaches a pre-defined milestone: total_iters.
```

File: torch/optim/lr_scheduler.py [hunk 3]
```diff
@@ -599,9 +607,11 @@ def _get_closed_form_lr(self):
 
 
 class SequentialLR(LRScheduler):
-    """Receives the list of schedulers that is expected to be called sequentially during
-    optimization process and milestone points that provides exact intervals to reflect
-    which scheduler is supposed to be called at a given epoch.
+    """
+    Receives the list of schedulers that is expected to be called sequentially during optimization process.
```
Comment (svekars) [inline torch/optim/lr_scheduler.py:611]:
```suggestion
    Receives the list of schedulers that is expected to be called sequentially during the optimization process.
```

**Comment by alperenunlu** [comment]
@svekars these suggested lines are too long for flake8 should I add them to the pr?

File: torch/optim/optimizer.py [hunk 4]
```diff
@@ -640,9 +645,11 @@ def register_load_state_dict_pre_hook(
         hook: Callable[["Optimizer", StateDict], Optional[StateDict]],
         prepend: bool = False,
     ) -> RemovableHandle:
-        r"""Register a load_state_dict pre-hook which will be called before
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
-        following signature::
+        r"""Register a load_state_dict pre-hook which will be called.
```
Comment (soulitzer) [inline torch/optim/optimizer.py:648]: This seems odd. Can you revert this change?

File: torch/optim/optimizer.py [hunk 5]
```diff
@@ -678,9 +685,11 @@ def register_load_state_dict_pre_hook(
     def register_load_state_dict_post_hook(
         self, hook: Callable[["Optimizer"], None], prepend: bool = False
     ) -> RemovableHandle:
-        r"""Register a load_state_dict post-hook which will be called after
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
-        following signature::
+        r"""Register a load_state_dict post-hook which will be called.
```
Comment (soulitzer) [inline torch/optim/optimizer.py:688]: Same here.

File: torch/optim/optimizer.py (hunk 4 above)
Comment (soulitzer) [reply torch/optim/optimizer.py:648]: You want to have the first sentence as:

Register a load_state_dict pre-hook which will be called before  :meth:`~torch.optim.Optimizer.load_state_dict` `is called 

(with the proper quoting)

File: torch/optim/optimizer.py [hunk 6]
```diff
@@ -837,7 +846,9 @@ def step(self, closure: Callable[[], float]) -> float:
         ...
 
     def step(self, closure: Optional[Callable[[], float]] = None) -> Optional[float]:
-        r"""Performs a single optimization step (parameter update).
+        r"""Perform a single optimization step.
+
+        Parameter update
```
Comment (soulitzer) [inline torch/optim/optimizer.py:851]: Please revert this change

File: torch/optim/lr_scheduler.py [hunk 7]
```diff
@@ -122,8 +120,10 @@ def print_lr(self, is_verbose, group, lr, epoch=None):
 
 
     def step(self, epoch=None):
-        # Raise a warning if old pattern is detected
-        # https://github.com/pytorch/pytorch/issues/20124
+        """Raise a warning if old pattern is detected.
```
Comment (soulitzer) [inline torch/optim/lr_scheduler.py:123]: We should want to this to be a comment, not a docstring

**Review by soulitzer — COMMENTED** [review]
Thanks for the fix, I added some suggestions (mainly where I think more context in the first sentence is important to understanding).

File: torch/optim/optimizer.py (hunk 4 above)
Comment (alperenunlu) [reply torch/optim/optimizer.py:648]: **pydocstyle** wants a summary line and linter wants the line shorter than 120 char.

>Register a load_state_dict pre-hook which will be called before :meth:`~torch.optim.Optimizer.load_state_dict` is called

still too long for linter to accept because of the indentation.

Comment (soulitzer) [reply torch/optim/optimizer.py:648]: I see. I think we should err on the of being more informative rather than appeasing the linter though.

Comment (alperenunlu) [reply torch/optim/optimizer.py:648]: @soulitzer I also felt like it became odd. But I couldn't come up with another solution that could pass by both pydocstyle and flake8.

Comment (soulitzer) [reply torch/optim/optimizer.py:648]: It is probably fine to leave it as it was before. That way flake8 passes, and we don't regress the meaning of the first sentence.

**Comment by alperenunlu** [comment]
Thank you @soulitzer!!

**Comment by alperenunlu** [comment]
@svekars Can you also look at soulitzer's review?

**Comment by soulitzer** [comment]
The autograd parts look good to me, but maybe @janeyx99 should take a closer look at the optim changes.

File: torch/optim/lr_scheduler.py [hunk 8]
```diff
@@ -85,21 +85,21 @@ def wrapper(*args, **kwargs):
...
+        """Return the state of the scheduler as a :class:`dict`.
 
         It contains an entry for every variable in self.__dict__ which
         is not the optimizer.
         """
         return {key: value for key, value in self.__dict__.items() if key != 'optimizer'}
 
     def load_state_dict(self, state_dict):
-        """Loads the schedulers state.
+        """Load the schedulers state.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:102]:
```suggestion
        """Load the scheduler's state.
```
Since we're at it, might as well fix these

File: torch/optim/lr_scheduler.py [hunk 9]
```diff
@@ -187,8 +185,9 @@ def __exit__(self, type, value, traceback):
 
 
 class LambdaLR(LRScheduler):
-    """Sets the learning rate of each parameter group to the initial lr
-    times a given function. When last_epoch=-1, sets initial lr as lr.
+    """Sets the learning rate of each parameter group to the initial lr times a given function.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:188]:
```suggestion
    """Set the learning rate of each parameter group to the initial lr times a given function.
```
Should this be singularized following the pattern everywhere else?

File: torch/optim/lr_scheduler.py [hunk 10]
```diff
@@ -276,8 +273,9 @@ def get_lr(self):
 
 
 class MultiplicativeLR(LRScheduler):
-    """Multiply the learning rate of each parameter group by the factor given
-    in the specified function. When last_epoch=-1, sets initial lr as lr.
+    """Multiply the learning rate of each parameter group by the factor given in the specified function.
+
+    When last_epoch=-1, sets initial lr as lr.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:278]:
```suggestion
    When last_epoch=-1, set the initial lr as lr.
```

File: torch/optim/lr_scheduler.py [hunk 11]
```diff
@@ -187,8 +185,9 @@ def __exit__(self, type, value, traceback):
 
 
 class LambdaLR(LRScheduler):
-    """Sets the learning rate of each parameter group to the initial lr
-    times a given function. When last_epoch=-1, sets initial lr as lr.
+    """Sets the learning rate of each parameter group to the initial lr times a given function.
+
+    When last_epoch=-1, sets initial lr as lr.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:190]:
```suggestion
    When last_epoch=-1, set initial lr as lr.
```
here too

Comment (janeyx99) [reply torch/optim/lr_scheduler.py:190]: maybe not, since this is a class..

File: torch/optim/lr_scheduler.py [hunk 12]
```diff
@@ -730,7 +738,7 @@ def state_dict(self):
         return state_dict
 
     def load_state_dict(self, state_dict):
-        """Loads the schedulers state.
+        """Load the schedulers state.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:741]:
```suggestion
        """Load the scheduler's state.
```

File: torch/optim/lr_scheduler.py [hunk 13]
```diff
@@ -801,8 +810,9 @@ def _get_closed_form_lr(self):
 
 
 class CosineAnnealingLR(LRScheduler):
-    r"""Set the learning rate of each parameter group using a cosine annealing
-    schedule, where :math:`\eta_{max}` is set to the initial lr and
+    r"""Set the learning rate of each parameter group using a cosine annealing schedule.
+
+    Where :math:`\eta_{max}` is set to the initial lr and
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:815]:
```suggestion
    :math:`\eta_{max}` is set to the initial lr and
```
Combine with following line

File: torch/optim/lr_scheduler.py [hunk 14]
```diff
@@ -878,9 +888,10 @@ def _get_closed_form_lr(self):
 
 
 class ChainedScheduler(LRScheduler):
-    """Chains list of learning rate schedulers. It takes a list of chainable learning
-    rate schedulers and performs consecutive step() functions belonging to them by just
-    one call.
+    """Chains list of learning rate schedulers.
+
+    It takes a list of chainable learning rate schedulers and performs
+    consecutive step() functions belonging to them by just one call.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:894]:
```suggestion
    their step() functions consecutively with just one call.
```

File: torch/optim/lr_scheduler.py [hunk 15]
```diff
@@ -934,7 +945,7 @@ def state_dict(self):
         return state_dict
 
     def load_state_dict(self, state_dict):
-        """Loads the schedulers state.
+        """Load the schedulers state.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:948]:
```suggestion
        """Load the scheduler's state.
```

File: torch/optim/lr_scheduler.py [hunk 16]
```diff
@@ -1376,8 +1388,9 @@ def load_state_dict(self, state_dict):
 
 
 class CosineAnnealingWarmRestarts(LRScheduler):
-    r"""Set the learning rate of each parameter group using a cosine annealing
-    schedule, where :math:`\eta_{max}` is set to the initial lr, :math:`T_{cur}`
+    r"""Set the learning rate of each parameter group using a cosine annealing schedule.
+
+    Where :math:`\eta_{max}` is set to the initial lr, :math:`T_{cur}`
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1393]:
```suggestion
    :math:`\eta_{max}` is set to the initial lr, :math:`T_{cur}`
```

File: torch/optim/optimizer.py [hunk 17]
```diff
@@ -640,11 +645,11 @@ def register_load_state_dict_pre_hook(
         hook: Callable[["Optimizer", StateDict], Optional[StateDict]],
         prepend: bool = False,
     ) -> RemovableHandle:
-        r"""Register a load_state_dict pre-hook which will be called before
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
+        r"""Register a state dict pre-hook which will be called before
+        :meth:`~torch.optim.Optimizer.state_dict` is called. It should have the
```
Comment (janeyx99) [inline torch/optim/optimizer.py:649]: These changes should be undone. load_state_dict is required to be mentioned, otherwise it will be confusing.

File: torch/optim/optimizer.py [hunk 18]
```diff
@@ -678,11 +683,11 @@ def register_load_state_dict_pre_hook(
     def register_load_state_dict_post_hook(
         self, hook: Callable[["Optimizer"], None], prepend: bool = False
     ) -> RemovableHandle:
-        r"""Register a load_state_dict post-hook which will be called after
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
+        r"""Register a state dict post-hook which will be called after
+        :meth:`~torch.optim.Optimizer.state_dict` is called. It should have the
```
Comment (janeyx99) [inline torch/optim/optimizer.py:687]: Same here.

File: torch/optim/optimizer.py [hunk 19]
```diff
@@ -640,11 +645,11 @@ def register_load_state_dict_pre_hook(
...
         prepend: bool = False,
     ) -> RemovableHandle:
-        r"""Register a load_state_dict pre-hook which will be called before
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
+        r"""Register a state dict pre-hook which will be called before
+        :meth:`~torch.optim.Optimizer.state_dict` is called. It should have the
         following signature::
 
-            hook(optimizer, state_dict) -> state_dict or None
+            hook(optimizer) -> None
```
Comment (janeyx99) [inline torch/optim/optimizer.py:652]: Wait..this is wrong. What inspires this change?

File: torch/optim/optimizer.py [hunk 20]
```diff
@@ -678,11 +683,11 @@ def register_load_state_dict_pre_hook(
...
         self, hook: Callable[["Optimizer"], None], prepend: bool = False
     ) -> RemovableHandle:
-        r"""Register a load_state_dict post-hook which will be called after
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
+        r"""Register a state dict post-hook which will be called after
+        :meth:`~torch.optim.Optimizer.state_dict` is called. It should have the
         following signature::
 
-            hook(optimizer) -> None
+            hook(optimizer, state_dict) -> state_dict or None
```
Comment (janeyx99) [inline torch/optim/optimizer.py:690]: This is wrong..what inspires this change?

**Review by janeyx99 — COMMENTED** [review]
Sorry for the delayed review, I've recently returned. 

Left my review--most edits are good but a few are wrong. Thank you for going through the docs!

File: torch/optim/lr_scheduler.py (hunk 9 above)
Comment (alperenunlu) [reply torch/optim/lr_scheduler.py:188]: Classes don't necessarily have to start with an imperative verb.
Therefore, I didn't modify the class docstrings to make them imperative.

File: torch/optim/lr_scheduler.py (hunk 11 above)
Comment (alperenunlu) [reply torch/optim/lr_scheduler.py:190]: I also didn't touch them because they are in a class and not the summary line.

File: torch/optim/lr_scheduler.py (hunk 10 above)
Comment (alperenunlu) [reply torch/optim/lr_scheduler.py:278]: I didn't touch them since they are in a class and not the summary line.

File: torch/optim/optimizer.py (hunk 19 above)
Comment (alperenunlu) [reply torch/optim/optimizer.py:652]: It seems I mixed them up while fixing pre and post. I am fixing them.

File: torch/optim/optimizer.py (hunk 20 above)
Comment (alperenunlu) [reply torch/optim/optimizer.py:690]: It seems I mixed pre and post. Fixing them.

File: torch/optim/optimizer.py (hunk 17 above)
Comment (alperenunlu) [reply torch/optim/optimizer.py:649]: You are right.

**Comment by alperenunlu** [comment]
Thank you for your review @janeyx99!!

**Review by janeyx99 — APPROVED** [review]
So sorry I did not get back to you earlier!

**Comment by janeyx99** [comment]
If you're willing to manually rebase and bring this PR up to date, please ping me and we can try to get this in this week.
